cylc/rose/entry_points.py

```diff
@@ -40,7 +41,15 @@ def pre_configure(srcdir: Path, opts: 'Values') -> dict:
         # nothing to do here
         return {}
 
-    # load the Rose config
+    # If we are validating against source, load saved CLI options
+    # from previous install, as saved in the rose-suite-cylc-install.conf
+    if (
+        getattr(opts, 'against_source', False)
+        and isinstance(opts.against_source, Path)
+    ):
+        opts = retrieve_installed_cli_opts(srcdir, opts)
```

Comment:
Should this have a `modify=False`?

Reply:
No - this function is written to be non-destructive (as you'd want the pre-configure plugin to be). When this function calls `record_cylc_install_options` it does pass `modify=False` to that. We haven't a use (at least not yet) for choosing to have a modifying version of this function.